### PR TITLE
Add forward and background arrows to canvas to undo/redo

### PR DESCRIPTION
**Pull request by OAGr**
These work, but there are some improvements to make in future releases:
- Going back/forward in history should save after some timeout (right now it does not result in something being saved.
- Clicking a deselected button can trigger highlight of surrounding text. 

image

### CODE REVIEW

**Comment by mmcdermott** [comment]
This ready to go in? I recall you saying it shouldn't be merged to master yet (but maybe you meant production?)

File: src/components/spaces/denormalized-space-selector.js [hunk 1]
```diff
@@ -1,8 +1,21 @@
 import { createSelector } from 'reselect';
 import e from 'gEngine/engine'
 
-const spaceGraphSelector = state => { return _.pick(state, 'spaces', 'metrics', 'guesstimates', 'simulations', 'users', 'organizations', 'userOrganizationMemberships', 'me') }
-const spaceSelector = (state, props) => {return state.spaces.find(s => s.id.toString() === props.spaceId.toString())}
+function _sameId(idA, idB){
+  return idA.toString() === idB.toString()
```
Comment (mmcdermott) [inline src/components/spaces/denormalized-space-selector.js]: Why do we need this defensivity here?

Comment (OAGr) [inline src/components/spaces/denormalized-space-selector.js]: We need it for at least one of the IDs, I forgot which.

File: src/components/spaces/show/index.js [hunk 2]
```diff
@@ -170,7 +185,11 @@ export default class SpacesShow extends Component {
                 actionState={space.canvasState.actionState}
                 canBePrivate={canBePrivate}
                 onPublicSelect={this.onPublicSelect.bind(this)}
-                onPrivateSelect={this.onPrivateSelect.bind(this)}
+                onPrivateSelecundot={this.onPrivateSelect.bind(this)}
+                onUndo={this.onUndo.bind(this)}
+                onRedo={this.onRedo.bind(this)}
+                canUndo={space.checkpointMetadata.head !== space.checkpointMetadata.length - 1}
```
Comment (mmcdermott) [inline src/components/spaces/show/index.js]: Your null state in the selector (head: 0, length: 0) sets canUndo to true. Maybe null state should be head: 0 length: 1?

File: src/components/spaces/show/header.js [hunk 3]
```diff
@@ -68,6 +76,17 @@ const SpaceHeader = ({
       </div>
 
       <div className='header-actions'>
+        {editableByMe &&
+          <div className='ui buttons tiny'>
+            <button onClick={onUndo} className={`ui icon button ${canUndo ? '' : 'disabled'}`} data-tip data-for='undo-button'>
+              <Icon name='undo'/>
+            </button>
+            <button onClick={onRedo} className={`ui icon button ${canRedo ? '' : 'disabled'}`} data-tip data-for='redo-button'>
+              <Icon name='rotate-right'/>
```
Comment (mmcdermott) [inline src/components/spaces/show/header.js]: This can also be called 'repeat' I believe.

**Pull request merged**